src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -124,26 +124,32 @@ public void consume(Message<String> message) {
                 }
 
                 if (!processConfigs.isEmpty()) {
-                    // import IGM into the network store
-                    UUID networkUuid = caseFetcherService.importCase(caseUuid);
-                    // check IGM quality
-                    boolean valid = igmQualityCheckService.check(networkUuid);
-
-                    merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid);
-
-                    for (ProcessConfig processConfig : processConfigs.subList(1, processConfigs.size())) {
-                        // import IGM into the network store
-                        UUID processConfigNetworkUuid = caseFetcherService.importCase(caseUuid);
-                        merge(processConfig, dateTime, date, tso, valid, processConfigNetworkUuid);
-                    }
+                    Mono<UUID> networkUuidMono = caseFetcherService.importCase(caseUuid);
+
+                    Mono<Boolean> validMono = networkUuidMono.flatMap(networkUuid ->
+                       // check IGM quality
+                        igmQualityCheckService.check(networkUuid)
+                    );
+
+                    validMono.zipWith(networkUuidMono, (valid, networkUuid) ->
+                            merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid).subscribe())
```

Comment:
why subscribe manually to merge() ? instead put it in the reactor chain

Reply:
here is the end of the chain of that branch .